client/data/marketplace/use-wpcom-plugins-query.ts

```diff
@@ -48,13 +48,13 @@ export const getWPCOMPluginsQueryParams = (
 	type: Type,
 	searchTerm?: string,
 	tag?: string
-): [ QueryKey, QueryFunction< any[] > ] => {
-	const cacheKey = getCacheKey( type + searchTerm + tag + '-normalized' );
-	const fetchFn = () =>
+): { queryKey: QueryKey; queryFn: QueryFunction< any[] > } => {
+	const queryKey = getCacheKey( type + searchTerm + tag + '-normalized' );
+	const queryFn = () =>
 		fetchWPCOMPlugins( type, searchTerm, tag ).then( ( data: { results: any[] } ) =>
 			normalizePluginsList( data.results )
 		);
-	return [ cacheKey, fetchFn ];
+	return { queryKey, queryFn };
```

Comment:
Noting that `getWPCOMPluginsQueryParams` is used here:

https://github.com/Automattic/wp-calypso/blob/8f1fa912356c59738ed483a536f84bb85f191ca1/client/my-sites/plugins/controller-logged-out.js#L39

and we might need to do some changes to accommodate the function signature difference.